src/main/java/uk/gov/cshr/report/factory/ReportRowFactory.java

```diff
@@ -89,4 +93,35 @@ public ModuleReportRow createModuleReportRow(CivilServant civilServant, Module m
 
         return reportRow;
     }
+
+    public CourseReportRow createCourseReportRow(CivilServant civilServant, Course course, CourseRecord courseRecord, Identity identity, boolean isProfessionReporter, boolean required) {
+        CourseReportRow reportRow = new CourseReportRow();
+
+        if (!isProfessionReporter) {
+            reportRow.setLearnerId(identity.getUid());
+            reportRow.setName(civilServant.getName());
+            reportRow.setEmail(identity.getUsername());
+        }
+
+        reportRow.setDepartment(civilServant.getOrganisation());
+        reportRow.setProfession(civilServant.getProfession());
+        reportRow.setOtherAreasOfWork(civilServant.getOtherAreasOfWork());
+        reportRow.setGrade(civilServant.getGrade());
+
+        reportRow.setCourseId(course.getId());
+        reportRow.setCourseTitle(course.getTitle());
+        reportRow.setCourseTopicId(course.getTopicId());
+        reportRow.setRequired(required);
+
+        String state = courseRecord.getState();
+        if (StringUtils.isBlank(state) || StringUtils.equalsIgnoreCase("null", state)) {
+            reportRow.setStatus("IN_PROGRESS");
```

Comment:
Can we create interface or enum with all the status?

Reply:
There is only one status COMPLETED which is stored in the database. As of now, IN_PROGRESS status is not re-used in the code anywhere but we can put it in a constant file if we re-use it.